runtime/drivers/athena/sql_store.go

```diff
@@ -0,0 +1,270 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse config: %w", err)
+	}
+
+	awsConfig, err := c.awsConfig(ctx, conf.AWSRegion)
+	if err != nil {
+		return nil, err
+	}
+
+	client := athena.NewFromConfig(awsConfig)
+	outputLocation, err := resolveOutputLocation(ctx, client, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	outputURL, err := url.Parse(outputLocation)
+	if err != nil {
+		return nil, err
+	}
+
+	// outputLocation s3://bucket/path
+	// unloadLocation s3://bucket/path/rill_tmp_<uuid>
+	// unloadPath path/rill_tmp_<uuid>
+	unloadFolderName := "rill_tmp_" + uuid.New().String()
+	bucketName := outputURL.Hostname()
+	unloadURL := outputURL.JoinPath(unloadFolderName)
+	unloadLocation := unloadURL.String()
+	unloadPath := strings.TrimPrefix(unloadURL.Path, "/")
+
+	cleanupFn := func() error {
+		return deleteObjectsInPrefix(ctx, awsConfig, bucketName, unloadPath)
+	}
+
+	err = c.unload(ctx, client, conf, unloadLocation)
+	if err != nil {
+		return nil, errors.Join(fmt.Errorf("failed to unload: %w", err), cleanupFn())
+	}
+
+	bucketObj, err := openBucket(ctx, awsConfig, bucketName)
+	if err != nil {
+		return nil, errors.Join(fmt.Errorf("cannot open bucket %q: %w", bucketName, err), cleanupFn())
+	}
+
+	opts := rillblob.Options{
+		GlobPattern: unloadPath + "/**",
+	}
+
+	it, err := rillblob.NewIterator(ctx, bucketObj, opts, c.logger)
+	if err != nil {
+		return nil, errors.Join(fmt.Errorf("cannot download parquet output %q %w", opts.GlobPattern, err), cleanupFn())
+	}
+
+	return autoDeleteFileIterator{
+		FileIterator: it,
+		cleanupFn:    cleanupFn,
+	}, nil
+}
+
+func (c *Connection) awsConfig(ctx context.Context, awsRegion string) (aws.Config, error) {
+	loadOptions := []func(*config.LoadOptions) error{
+		// Setting the default region to an empty string, will result in the default region value being ignored
+		config.WithDefaultRegion("us-east-1"),
+		// Setting the region to an empty string, will result in the region value being ignored
+		config.WithRegion(awsRegion),
+	}
+
+	// If one of the static properties is specified: access key, secret key, or session token, use static credentials,
+	// Else fallback to the SDK's default credential chain (environment, instance, etc) unless AllowHostAccess is false
+	if c.config.AccessKeyID != "" || c.config.SecretAccessKey != "" || c.config.SessionToken != "" {
+		p := credentials.NewStaticCredentialsProvider(c.config.AccessKeyID, c.config.SecretAccessKey, c.config.SessionToken)
+		loadOptions = append(loadOptions, config.WithCredentialsProvider(p))
+	} else if !c.config.AllowHostAccess {
+		return aws.Config{}, fmt.Errorf("static creds are not provided, and host access is not allowed")
+	}
+
+	return config.LoadDefaultConfig(ctx, loadOptions...)
+}
+
+func (c *Connection) unload(ctx context.Context, client *athena.Client, conf *sourceProperties, unloadLocation string) error {
+	finalSQL := fmt.Sprintf("UNLOAD (%s) TO '%s' WITH (format = 'PARQUET')", conf.SQL, unloadLocation)
```

Comment:
We should put a newline after the injected SQL (`UNLOAD (%s\n) ...`) in case there's a comment at the end of it